ArchiSteamFarm/Core/Utilities.cs

```diff
@@ -319,6 +327,42 @@ public static class Utilities {
 			}
 		}
 
+		internal static bool IsWeakPassword(string password, ISet<string>? additionallyForbiddenPhrases = null) {
+			if (string.IsNullOrEmpty(password)) {
+				throw new ArgumentNullException(nameof(password));
+			}
+
+			HashSet<string> forbiddenPhrases = ForbiddenPasswordPhrases.ToHashSet(StringComparer.InvariantCultureIgnoreCase);
+
+			if (additionallyForbiddenPhrases != null) {
+				forbiddenPhrases.UnionWith(additionallyForbiddenPhrases);
+			}
+
+			int remainingCharacters = password.Length;
+
+			foreach (string forbiddenPhrase in forbiddenPhrases.Where(static phrase => !string.IsNullOrEmpty(phrase))) {
+				for (int index = password.IndexOf(forbiddenPhrase, StringComparison.InvariantCultureIgnoreCase); index >= 0; index = password.IndexOf(forbiddenPhrase, index, StringComparison.InvariantCultureIgnoreCase)) {
+					remainingCharacters -= forbiddenPhrase.Length - 1;
+
+					if (remainingCharacters < MinimumRecommendedPasswordCharacters) {
+						return true;
+					}
+				}
+			}
+
+			for (int i = 2; i < password.Length; i++) {
+				ushort ch0 = password[i - 2];
+				ushort ch1 = password[i - 1];
+				ushort ch2 = password[i];
+
+				if (ch0 == ch2 && ch1 == ch2 || ch0 == ch2 - 2 && ch1 == ch2 - 1 || ch0 == ch2 + 2 && ch1 == ch2 + 1) {
+					return true;
```

Comment:
I don't think we can assume that password containing 3 repeating or sequential chars is a weak password, example: `definitelyDifficultPassword` - this will be marked as weak because `def` is sequential. Consider decrementing `remainingCharacters`, that would make `aaabcdefg` 2 characters long, which is essentially weak password.

Reply:
Either that or increasing the threshold. Let's wait for the opinion of @JustArchi

Reply:
Decrementing the characters sounds better to me, because `abcdefg` will be counted as `1`.

Alternatively check all the words in english language to find out the lowest `N` value that can be used for sequential threshold which still doesn't produce any false-positives. That might be `4`, or more.

And maybe we should do both actually.

Reply:
That means tho, that we need to loop through the password three times, because we need to find the length of the unwanted pattern...

Reply:
Why? Store first char as lastChar, count = 0, foreach char in chars starting from second, compare with lastChar, if +1 or -1, set lastChar to that and increment count. If not, remove count from length, then re-set lastChar to current and count to 0.

`abcdef` will therefore have length of `1`, `definitely` will have count of 7.

Works both ways, and as an extra you cover `abcdcba`.

Reply:
Then once you have above, add logic to remove only counts > 3, so casual `def` won't be counted whatsoever, while `abcdef` will.

Reply:
Also we probably should care about passwords like ’qwertyuiop’

Reply:
Well. I didn't think of bringing `abba` to the mix 😉 With that added: I agree that we only need to loop once. 

Should we remove the whole count or continue the same way as with our banned words and count them as just 1 char?

Reply:
> Also we probably should care about passwords like ’qwertyuiop’

In theory I'd agree, but if we start doing that we should probably also check for the other key rows ANd for the most common other keyboard layouts (such as `qertz` or `azerty`)

Reply:
I'd say continue, `abcXdef` should still be `3` and not `1`.

Reply:
> In theory I'd agree, but if we start doing that we should probably also check for the other key rows AND for the most common other keyboard layouts (such as qertz or azerty). And of course the reverse of all of those -.-

Check the library Vital recommended, maybe we're doing foolish job after all here trying to guess if their algo is much better.

Reply:
I'll do that some time after waking up. It looks sophisticated at first glance, but *might* be overkill.